### ISSUE TITLE
Add org.apache.sling.servlets.annotations to dependencyManagement

### ISSUE THREAD

**Issue by kwin** [adobe/aem-project-archetype#340]
Nowadays servlets should be registered via the annotations outlined in https://sling.apache.org/documentation/the-sling-engine/servlets.html#registering-a-servlet-using-java-annotations. That requires the dependency 
```
<artifactId>org.apache.sling.servlets.annotations</artifactId>
<groupId>org.apache.sling</groupId>
```

That dependency should be managed in the root pom and added to the dependencies of the bundles.

**Comment by bpauli** [comment]
@kwin Thank you for this enhancement request.

**Comment by bpauli** [comment]
Postpone as the `org.apache.sling.servlets.annotations` artifact is currently not exposed by the `aem-sdk-api`. Created an internal issue GRANITE-29022 to change this. Once this is done we can merge the PR.

**Comment by kwin** [comment]
As this is a build time artifact only, I would strongly recommend to not put it into the aem-sdk-app. It should be updated to the latest version independent of the AEM runtime version.

**Comment by bpauli** [comment]
As the `aem-sdk-api` is daily release this shouldn't be a problem anymore. We'll take care to keep the annotations up to date.

**Comment by cziegeler** [comment]
The AEM SDK defines which annotations should be used for AEM, so must be part of the SDK - it has nothing to do with runtime. As Burkhard points out we release daily and make sure to maintain the version. We also need to control that the version used matches what is deployed and not ahead or behind.

**Comment by kwin** [comment]
> We also need to control that the version used matches what is deployed and not ahead or behind.
The servlet annotations do not depend on a specific runtime bundle. In general the most recent version works with all runtimes (but for individual features it requires a certain runtime version of servlet resolver).  Still not updating to the most recent version of the annotations which supports a not yet supported feature sounds wrong to me, as the most recent version of the annotations may contain other bugfixes as well.

**Comment by cziegeler** [comment]
Again, as stated, we will keep this up to date. 
And as the annotation version is not hard coded in the project anymore, we can ensure that the latest is used without requiring to update individual dependencies.
But as you also state, newer annotations might require newer runtime features (like in the servlet resolver). We might not have updated the runtime with that. That's why we make sure that the best possible version is used and don't leave it to our users.

**Comment by kwin** [comment]
> Again, as stated, we will keep this up to date.

Unfortunately the version used in the most recent https://repo1.maven.org/maven2/com/adobe/aem/aem-sdk-api/2021.11.6058.20211123T163652Z-211000/aem-sdk-api-2021.11.6058.20211123T163652Z-211000.pom is outdated. It still references 1.2.4, but 1.2.6 was released in June.

@cziegeler Can you update that?

**Comment by cziegeler** [comment]
@kwin Thanks for the reminder, yes, will do
